include/xtensor/xcomplex.hpp

```diff
@@ -129,5 +129,86 @@ namespace xt
     {
         return detail::complex_expression_helper<is_xexpression<std::decay_t<E>>::value>::imag(std::forward<E>(e));
     }
+
+#define UNARY_COMPLEX_FUNCTOR(NAME)\
+    template <class T>\
+    struct NAME##_fun {\
+        using argument_type = T;\
+        using result_type = decltype(std::NAME(std::declval<T>()));\
+        constexpr result_type operator()(const T& t) const {\
+            using std::NAME;\
+            return NAME(t);\
+        }\
+    }
+
+    namespace math
+    {
+        const double PI  = 3.141592653589793238463;
```

Comment:
Rather than a hard-coded constant, I think that a `xt::numerical_values` template class (similar to `std::numerical_limits`) holding static universal constants would make sense.